db/import_listener.go

```diff
@@ -55,8 +55,11 @@ func (il *importListener) StartImportFeed(bucket base.Bucket, dbStats *base.DbSt
 
 	importFeedStatsMap := dbContext.DbStats.Database().ImportFeedMapStats
 
-	// Register cbgt PIndex to support sharded import.
-	il.RegisterImportPindexImpl()
+	// Store the listener in global map for dbname-based retrieval by cbgt prior to index registration
+	base.StoreDestFactory(base.ImportDestKey(il.database.Name), il.NewImportDest)
+
+	// Register cbgt PIndex to support sharded import, if not already defined
+	//il.RegisterImportPindexImpl()
```

Comment:
Looks like this is handled elsewhere now via the globally scoped `RegisterImportPindexImpl` function.